src/docfx/lib/github/GitHubUserCache.cs

```diff
@@ -150,6 +154,27 @@ public async Task<Error> SaveChanges(Config config)
                 return null;
             }
 
+            var remainingRetries = 30;
```

Comment:
Is `30` too big? Upload failure is not catastrophic, it does not affect output correctness. We typically only retry 3 times for get operations, maybe just retry 3 times here. Let's say when the server is buzy and each request timed out. Default HttpClient timeout is 100s, it will take at least 3000s (~1 hour) for this operation to finish.